app/src/main/java/com/android/streetworkapp/model/workout/PairingRequest.kt

```diff
@@ -0,0 +1,25 @@
+package com.android.streetworkapp.model.workout
+
+/**
+ * A data class representing a pairing request between users.
+ *
+ * @param requestId Unique ID of the pairing request (generated by the database or app).
+ * @param fromUid UID of the user sending the request.
+ * @param toUid UID of the user receiving the request (e.g., a potential coach).
+ * @param status The current status of the request (e.g., PENDING, ACCEPTED, REJECTED).
+ * @param timestamp The time when the request was created, in epoch milliseconds.
+ */
+data class PairingRequest(
```

Comment:
Question: I am not sure I understand how the request will be implemented later in the UI: will the user need to share their User ID?

Reply:
The idea behind it is that the coach sends a request with `sendPairingRequest` and the athele observes the requests that are in his name, when he finds one he can either accept it or refuse it. If he accepts, the session begins.

Reply:
"observes the requests that are in his name" : What do you mean by "in his name"? If I want to start coaching Bob, how do I send a request to Bob? Do I need to know his Name? If yes, will I have to write his name in the app to start the coaching? Or do you imagine a different kind of user experience?

Reply:
By name is the uid attribut inside the `PairingRequest` (my bad for this shortcut)

The procedure:
1. The coach (Bob) create a pairing request by choosing one of his friends
2. The athlete observe the requests that contains his uid (+ in pending status)
3. The athlete accepts/refuses the request. 

-> No name writing required
-> Just a list of friends (coach side), and list of requests (athlete side)

Hopefully this will clarify your questions.

Reply:
Thank you for the clarification, I did not understand that the users must be friend before they can coach each other. It seem like a very nice user experience!